riscv/tests/riscv_data/vec_median/src/lib.rs

```diff
@@ -0,0 +1,90 @@
+//! Calculates the round-down median of a given input vector, and compares with
+//! the expected value.
+//!
+//! First argument is the expected value, second is the number of elements, the
+//! other are the elements.
+//!
+//! For example, the following will calculate the median of the 11 elements
+//! vector [15,75,6,5,1,4,7,3,2,9,2] and compare the result with the expected
+//! value of 5:
+//! ```
+//! cargo run --release rust riscv/tests/riscv_data/vec_median -o tmp -f -i 5,11,15,75,6,5,1,4,7,3,2,9,2
+//! ```
+
+#![no_std]
+
+extern crate alloc;
+use core::{
+    alloc::{GlobalAlloc, Layout},
+    cell::Cell,
+    ptr::{self, addr_of},
+};
+
+use alloc::vec::Vec;
+use runtime::get_prover_input;
+
+struct VeryDumbAllocator<const SIZE: usize> {
+    mem_buffer: [u8; SIZE],
+    next_available: Cell<usize>,
+}
+
+impl<const SIZE: usize> VeryDumbAllocator<SIZE> {
+    const fn new() -> Self {
+        Self {
+            mem_buffer: [0; SIZE],
+            next_available: Cell::new(0),
+        }
+    }
+}
+
+unsafe impl<const SIZE: usize> GlobalAlloc for VeryDumbAllocator<SIZE> {
+    unsafe fn alloc_zeroed(&self, layout: Layout) -> *mut u8 {
+        // Start address of the allocation array:
+        let array_start = addr_of!(self.mem_buffer) as usize;
+
+        // Address of the next free space:
+        let next_ptr = array_start + self.next_available.get();
+
+        // Align the pointer.
+        let next_aligned_ptr = ((next_ptr - 1) / layout.align() + 1) * layout.align();
+
+        // Where this allocated space ends:
+        let end_of_allocation = next_aligned_ptr + layout.size();
+
+        if end_of_allocation - array_start <= SIZE {
+            self.next_available.set(end_of_allocation);
+            next_aligned_ptr as *mut u8
+        } else {
+            ptr::null_mut()
+        }
+    }
+
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        self.alloc_zeroed(layout)
```

Comment:
I think we should support allocating non-zeroed data. Zeroing is very expensive.

Reply:
Ah sorry, - it is all zeroed :)